src/Agent.Listener/SelfUpdater.cs

```diff
@@ -76,13 +77,16 @@ public async Task<bool> SelfUpdate(AgentRefreshMessage updateMessage, IJobDispat
 
             // kick off update script
             Process invokeScript = new Process();
-#if OS_WINDOWS
-            invokeScript.StartInfo.FileName = WhichUtil.Which("cmd.exe", trace: Trace);
-            invokeScript.StartInfo.Arguments = $"/c \"{updateScript}\"";
-#elif (OS_OSX || OS_LINUX)
-            invokeScript.StartInfo.FileName = WhichUtil.Which("bash", trace: Trace);
-            invokeScript.StartInfo.Arguments = $"\"{updateScript}\"";
-#endif
+            if (PlatformUtil.RunningOnWindows)
+            {
+                invokeScript.StartInfo.FileName = WhichUtil.Which("cmd.exe", trace: Trace);
+                invokeScript.StartInfo.Arguments = $"/c \"{updateScript}\"";
+            }
+            else if (PlatformUtil.RunningOnMacOS || PlatformUtil.RunningOnLinux)
```

Comment:
why not just else?

Reply:
No good reason. I think I was on the path of aping exactly the logic that was previously present. I'll fix.